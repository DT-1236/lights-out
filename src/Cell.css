.Cell {
  height: 100px;
  width: 100px;
  background-color: darkgray;
  outline: 1px black solid;
}

.Cell-lit {
  background-color: white;
}
