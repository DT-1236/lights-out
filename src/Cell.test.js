import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Cell from './Cell';

it('renders without crashing', () => {
  shallow(<Cell />);
});

it('matches the lit snapshot', () => {
  const wrapper = mount(
    <table>
      <tbody>
        <tr>
          <Cell isLit={true} />
        </tr>
      </tbody>
    </table>
  );
  const serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

it('matches the off snapshot', () => {
  const wrapper = mount(
    <table>
      <tbody>
        <tr>
          <Cell isLit={false} />
        </tr>
      </tbody>
    </table>
  );
  const serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
