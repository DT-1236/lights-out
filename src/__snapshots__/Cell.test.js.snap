// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the lit snapshot 1`] = `
<table>
  <tbody>
    <tr>
      <Cell
        isLit={true}
      >
        <td
          className="Cell Cell-lit"
          onClick={[Function]}
        />
      </Cell>
    </tr>
  </tbody>
</table>
`;

exports[`matches the off snapshot 1`] = `
<table>
  <tbody>
    <tr>
      <Cell
        isLit={false}
      >
        <td
          className="Cell"
          onClick={[Function]}
        />
      </Cell>
    </tr>
  </tbody>
</table>
`;
